### PR TITLE
Fix/#7099: add `exclude_ignored_assets` in trades and asset_movements query

### LINKED ISSUE #7099: Trades and Deposits/withdrawals should not return events with ignored assets
## Problem Definition

At the moment the history events API towards which we want to go is by default not returning any events that contain an ignored_asset. This is tweakable behaviour via the `exclude_ignored_assets` API argument but it's always set to true.

We also have events in the old format in the trades API endpoing and the asset movements API endpoint. Those endpoints do not follow the same logic. They will return all events despite containing an ignored asset.

The plan is to move trades and deposit/withdrawals into history events and essentially remove those api endpoints, but until that happens we need to have common functionality between them and not show to the user events with ignored assets.

## Task

Backend: Add same functionality we have in the history events endpoint to trades and deposit/withdrawals. Essentially a default true `exclude_ignored_assets` argument. Add it to the filter and make sure ignored asset events are skipped. Add tests.
Frontend: Make it tweakable.

### PR DESCRIPTION
**Pull request by OjusWiZard**
Closes #7099

## Checklist
- [x] add `exclude_ignored_assets` query parameter in trades and asset_movements query
- [x] add their tests
- [x] update the API docs

### CODE REVIEW

File: rotkehlchen/tests/api/test_exchanges.py [hunk 1]
```diff
@@ -801,6 +803,46 @@ def assert_okay(response):
...
+        without_ignore_response = assert_proper_response_with_result(without_ignore_response)
+
+        toggle_ignore_an_asset(rotkehlchen_api_server_with_exchanges, A_BTC)
+        data |= {'exclude_ignored_assets': True}
+        ignored_response = requests.get(api_url_for(server, 'assetmovementsresource'), json=data)
+        ignored_response = assert_proper_response_with_result(ignored_response)
+        default_response = requests.get(api_url_for(server, 'assetmovementsresource'))
+        default_response = assert_proper_response_with_result(default_response)
+        msg = '`exclude_ignored_assets` should be True by default'
+        assert ignored_response == default_response, msg
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_exchanges.py:822]: nitipck. Keep things compact here and in other places in the code

```suggestion
        assert ignored_response == default_response, '`exclude_ignored_assets` should be True by default'  # noqa: E501
```

File: rotkehlchen/tests/utils/accounting.py [hunk 2]
```diff
@@ -345,3 +350,47 @@ def assert_csv_export(
                 expected_csv_data=expected_csv_data,
                 expected_pnls=expected_pnls,
             )
+
+
+def toggle_ignore_an_asset(
+        rotkehlchen_api_server: APIServer,
+        asset_to_ignore: Asset,
+) -> None:
+    """Utility functions to add/remove an ignored asset"""
```
Comment (LefterisJP) [inline rotkehlchen/tests/utils/accounting.py:359]:
```suggestion
    """Utility function to add/remove an ignored asset"""
```

File: rotkehlchen/tests/utils/accounting.py [hunk 3]
```diff
@@ -345,3 +350,47 @@ def assert_csv_export(
...
+                'ignoredassetsresource',
+            ), json={'assets': [asset_to_ignore.identifier]},
+        )
+        assert response.status_code == 200
+        response_result = response.json().get('result', [])
+        assert asset_to_ignore.identifier in response_result
+
+        with rotki.data.db.conn.read_ctx() as cursor:
+            result = rotki.data.db.get_ignored_asset_ids(cursor)
+        assert asset_to_ignore.identifier in result
```
Comment (LefterisJP) [inline rotkehlchen/tests/utils/accounting.py:396]: Can also do this for readability and compactness

```suggestion
    return
    
    # else add the asset to the ignored list
    response = requests.put(
        api_url_for(
            rotkehlchen_api_server,
            'ignoredassetsresource',
        ), json={'assets': [asset_to_ignore.identifier]},
    )
    assert response.status_code == 200
    response_result = response.json().get('result', [])
    assert asset_to_ignore.identifier in response_result

    with rotki.data.db.conn.read_ctx() as cursor:
        result = rotki.data.db.get_ignored_asset_ids(cursor)
    assert asset_to_ignore.identifier in result
```

File: rotkehlchen/tests/utils/history.py [hunk 4]
```diff
@@ -736,6 +736,81 @@ def assert_kraken_asset_movements(
     assert_asset_movements(expected, to_check_list, deserialized, movements_to_check)
 
 
+def assert_ignored_btc_asset_movements(
```
Comment (LefterisJP) [inline rotkehlchen/tests/utils/history.py:739]: do not do it like this. If it's not used in at least 2 places do not create a new function.

File: rotkehlchen/tests/api/test_exchanges.py (hunk 1 above)
Comment (OjusWiZard) [reply rotkehlchen/tests/api/test_exchanges.py:822]: Oh I was trying to avoid `# noqa: E501`, but okay.

Comment (LefterisJP) [reply rotkehlchen/tests/api/test_exchanges.py]: yeah I get it. Some times for compactness it's okay to have it.

**Review by LefterisJP — APPROVED** [review]
LGTM.

Only thing missing is a changelog entry as this has a subtle but meaningful change for users. Can add later if at all

**Pull request merged**